packages/js/product-editor/src/blocks/generic/toggle/edit.tsx

```diff
@@ -64,13 +64,26 @@ export function Edit( {
 		}
 	}
 
+	/**
+	 * Create a Help component in case
+	 * the help attribute is set.
+	 */
+	const help = attributes?.help
+		? sanitizeHTML( attributes.help )?.__html
+		: null;
+
+	const helpComponent = createElement( 'div', {
+		dangerouslySetInnerHTML: { __html: help },
+	} );
+
 	return (
 		<div { ...blockProps }>
 			<ToggleControl
 				label={ label }
 				checked={ isChecked() }
 				disabled={ disabled }
 				onChange={ handleChange }
+				help={ isChecked() ? null : helpComponent }
```

Comment:
I think we should go with the more scalable approach, as this logic is very specific for a generic `toggle` block/field.